Sources/IRGen/Function/IRStatement.swift

```diff
@@ -37,145 +39,130 @@ struct IRStatement {
 struct IRIfStatement {
   var ifStatement: IfStatement
 
-  func rendered(functionContext: FunctionContext) -> String {
-    let condition0 = IRExpression(expression: ifStatement.condition).rendered(functionContext: functionContext)
+  func rendered(functionContext: FunctionContext) -> YUL.Statement {
+    let condition = IRExpression(expression: ifStatement.condition).rendered(functionContext: functionContext)
 
     let functionContext = functionContext
     functionContext.scopeContext = ifStatement.ifBodyScopeContext!
 
-    let body = ifStatement.body.map { statement in
-      return IRStatement(statement: statement).rendered(functionContext: functionContext)
-      }.joined(separator: "\n")
-    let ifCode: String
-
-    // TODO fixme for using some weird code
-    ifCode = """
-    \(condition0.preamble)
-    switch \(condition0.expression)
-    case 1 {
-      \(body.indented(by: 2))
+    let body = functionContext.withNewBlock {
+      ifStatement.body.forEach { statement in
+        functionContext.emit(IRStatement(statement: statement).rendered(functionContext: functionContext))
+      }
     }
-    """
-
-    var elseCode = ""
 
     if !ifStatement.elseBody.isEmpty {
       functionContext.scopeContext = ifStatement.elseBodyScopeContext!
-      let body = ifStatement.elseBody.map { statement in
-        if case .returnStatement(_) = statement {
-          fatalError("Return statements in else blocks are not supported yet")
+      let elseBody = functionContext.withNewBlock {
+        ifStatement.elseBody.forEach { statement in
+          functionContext.emit(IRStatement(statement: statement).rendered(functionContext: functionContext))
         }
-        return IRStatement(statement: statement).rendered(functionContext: functionContext)
-        }.joined(separator: "\n")
-      elseCode = """
-      default {
-        \(body.indented(by: 2))
       }
-      """
+      return .switch(Switch(condition, cases: [(YUL.Literal.num(1), body)], default: elseBody))
     }
 
-    return ifCode + "\n" + elseCode
+    return .switch(Switch(condition, cases: [(YUL.Literal.num(1), body)]))
   }
 }
 
 /// Generates code for a for statement.
 struct IRForStatement {
   var forStatement: ForStatement
 
-  func rendered(functionContext: FunctionContext) -> String {
+  func rendered(functionContext: FunctionContext) -> YUL.Statement {
     let functionContext = functionContext
     functionContext.scopeContext = forStatement.forBodyScopeContext!
 
-    let setup: String
-
     switch forStatement.iterable {
     case .identifier(let arrayIdentifier):
-      setup = generateArraySetupCode(prefix: "flint$\(forStatement.variable.identifier.name)$",
-        iterable: arrayIdentifier, functionContext: functionContext)
+      return .`for`(generateArraySetupCode(prefix: "flint$\(forStatement.variable.identifier.name)$",
```

Comment:
You don't need to backtick escape when using the case, only at the declaration site (iirc)